R/disability.R

```diff
@@ -0,0 +1,181 @@
+#' developmental disability as defined by
+#' [Brown et al., 2021](https://jamanetwork.com/journals/jamanetworkopen/fullarticle/2776018).
+#'
+#' Broadly, this includes any diagnosis codes for conditions that are likely to
+#' result in functional limitations and are considered to be chronic (see
+#' Supplemental eTable 2 in [Brown et al., 2021](https://jamanetwork.com/journals/jamanetworkopen/fullarticle/2776018)
+#' for a full list of diagnosis codes). However, note that this function only
+#' searches for relevant diagnosis codes at the encounter level and does not
+#' check whether chronic conditions are coded consistently across different
+#' encounters for a given patient.
+#'
+#' By setting `component_wise` to `TRUE`, users can choose to return all
+#' identified diagnosis codes for encounters with a disability, together with
+#' the disability category each code was matched with (e.g., physical/sensory/
+#' developmental disability - see below).
+#'
+#' @param cohort (`data.frame` or `data.table`)
+#' Cohort table with all relevant encounters of interest, where each row
+#' corresponds to a single encounter. Must contain GEMINI Encounter ID
+#' (`genc_id`).
+#'
+#' @param ipdiag (`data.table`)
+#' `ipdiagnosis` table as defined in the [GEMINI Data Repository Dictionary](https://drive.google.com/uc?export=download&id=1iwrTz1YVz4GBPtaaS9tJtU0E9Bx1QSM5).
+#' This table must contain `genc_id` and `diagnosis_code` (as ICD-10-CA
+#' alphanumeric code) in long format.
+#'
+#' @param erdiag (`data.table`)
+#' `erdiagnosis` table as defined in the [GEMINI Data Repository Dictionary](https://drive.google.com/uc?export=download&id=1iwrTz1YVz4GBPtaaS9tJtU0E9Bx1QSM5).
+#' This table must contain `genc_id` and `er_diagnosis_code` (as ICD-10-CA
+#' alphanumeric code) in long format.
+#' Typically, ER diagnoses should be included when deriving disability in order
+#' to increase sensitivity. However, in certain scenarios, users may choose to
+#' only include IP diagnoses by specifying `erdiag = NULL`. This may be useful
+#' when comparing cohorts with different rates of ER admissions.
+#'
+#' @param component_wise (`logical`)
+#' If `component_wise == FALSE` (default), only a single (global) disability
+#' flag will be returned indicating whether each `genc_id` was diagnosed with
+#' *any* disability (`FALSE/TRUE/NA`).
+#'
+#' If `component_wise == TRUE`, for each `genc_id` with a disability,
+#' all identified disability diagnosis codes are returned in long format. An
+#' additional column `disability_category` is returned containing
+#' one of the following 7 disability categories for each diagnosis:
+#' - Physical disability - Congenital Anomalies
+#' - Physical disability - Musculoskeletal disorders
+#' - Physical disability - Neurological disorders
+#' - Physical disability - Permanent Injuries
+#' - Sensory disabilities - Hearing impairments
+#' - Sensory disabilities - Vision impairments
+#' - Developmental Disabilities
+#'
+#' @section Warning:
```

Comment:
Maybe call the section "Notes" instead of "Warning". For the frailty_score function, I dedicated warning sections for documenting important warning messages the function produces. Nothing crucial, just for consistency in terms of tag uses.